adsdata/metrics.py

```diff
@@ -96,13 +97,20 @@ def read(self, db_conn, bibcode):
         first = r.first()
         return first
 
+    def get_by_bibcode(self, session, bibcode):
+        first = session.query(models.MetricsTable).filter(models.MetricsTable.bibcode==bibcode).first()
+        return first
+
+
     def get_by_bibcodes(self, db_conn, bibcodes):
         """ return a list of metrics datbase objects matching the list of passed bibcodes"""
         Session = sessionmaker()
         sess = Session(bind=db_conn)
-        x = sess.execute(select([self.table], self.table.c.bibcode.in_(bibcodes))).fetchall()
+        query = sess.query(models.MetricsTable).filter(models.MetricsTable.bibcode.in_(bibcodes))
```

Comment:
this will likely cause a problem; the session is closed and when you try to access the objects, sqa will complain - the previous code worked because no ORM was involved. Am I missing something?

Reply:
changed to use session, like other functions.